mxcubecore/HardwareObjects/abstract/ISPyBAbstractLims.py

```diff
@@ -129,6 +129,9 @@ def ldap_login(self, login_name, psd):
 
         return self.ldapConnection.authenticate(login_name, psd)
 
+    def ispyb_login(self, login_name, psd):
+        raise Exception("Abstract class. Not implemented")
```

Comment:
```suggestion
        raise NotImplementedError
```

https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
This is the style used in this class, but I agree with @fabcor-maxiv, it should raise `NotImplementedError`. I'll change it everywhere.

Also, it should be:

`raise NotImplementedError()`

😅

Reply:
I read somewhere that `raise SomeException` ends up running faster in some cases but never slower than `raise SomeException()`. There is also this linting rule: <https://docs.astral.sh/ruff/rules/unnecessary-paren-on-raise-exception/>.

Anyway, with or without parenthesises, my point is that we should use `NotImplementedError`.

Reply:
Changed to raise `NotImplementedError` everywhere.